src/lib.rs

```diff
@@ -24,3 +24,214 @@ pub use halo2_proofs::{
     self, halo2curves,
     halo2curves::{ff, group},
 };
+
+pub mod prelude {
+    use std::num::NonZeroUsize;
+
+    use crate::ff::{FromUniformBytes, PrimeFieldBits};
+    pub use crate::{
+        commitment::CommitmentKey,
+        ff::PrimeField,
+        ivc::{StepCircuit, IVC},
+    };
+
+    /// Within the IVC framework, on-circuit & off-circuit random oracle will be used
+    ///
+    /// This type defines this pair
+    ///
+    /// Only poiseidon is currently available
+    pub type RandomOracle<const T: usize, const RATE: usize> = super::poseidon::PoseidonRO<T, RATE>;
+
+    /// Within the IVC framework, on-circuit & off-circuit random oracle will be used
+    ///
+    /// This type defines constant what needed for create random oracle
+    ///
+    /// Only poiseidon is currently available
+    pub type RandomOracleConstant<F, const T: usize, const RATE: usize> =
+        <RandomOracle<T, RATE> as super::poseidon::ROPair<F>>::Args;
+
+    /// The IVC uses big uint math on-circuit and this parameter allows you to configure
+    /// the limb width of limb
+    ///
+    /// Safety: because 32 != 0
+    pub const DEFAULT_LIMB_WIDTH: NonZeroUsize = unsafe { NonZeroUsize::new_unchecked(32) };
+
+    /// The IVC uses big uint math on-circuit and this parameter allows you to configure
+    /// the maximum number of limbs allowed
+    ///
+    /// Safety: because 10 != 0
+    pub const DEFAULT_LIMBS_COUNT_LIMIT: NonZeroUsize = unsafe { NonZeroUsize::new_unchecked(10) };
+
+    pub const DEFAULT_STEP_FOLDING_CIRCUIT_SIZE: usize = 5;
```

Comment:
although prelude is to simplify tutorial example, not sure if we should setup this constant here

Reply:
No, the prelude is not to simplify the example, but to simplify the use of crate in general

All these constants are not important at the beginning of development, we don't even have instructions on how to select them optimally. So it is quite acceptable to choose default values and just describe how to change them if needed.